glib/src/variant.rs

```diff
@@ -1230,8 +1234,13 @@ impl FromVariant for std::ffi::OsString {
 #[cfg(windows)]
 impl FromVariant for std::ffi::OsString {
     fn from_variant(variant: &Variant) -> Option<Self> {
-        let wide = <Vec<u16>>::from_variant(variant)?;
-        Some(std::os::windows::ffi::OsStringExt::from_wide(&wide))
+        // Remove the ending NUL like with the Unix variant.
+        let mut bytes = <Vec<u8>>::from_variant(variant)?;
+        while bytes.last() == Some(&0) {
+            bytes.pop();
+        }
+        let s = String::from_utf8_lossy(&bytes);
```

Comment:
I.e. here you would panic instead

Reply:
Makes sense, I'll use `c_to_path_buf` and `c_to_os_string` instead `PathBuf::FromVariant` calling `OsString::FromVariant`.